src/FSharpVSPowerTools.Core/Utils.fs

```diff
@@ -545,14 +700,18 @@ module Dict =
 [<RequireQualifiedAccess>]
 [<CompilationRepresentation(CompilationRepresentationFlags.ModuleSuffix)>]
 module String =
+
+    let inline toCharArray (str:string) = str.ToCharArray()
+
     let lowerCaseFirstChar (str: string) =
-        match str with
-        | null -> null
-        | str ->
-            match str.ToCharArray() |> Array.toList with
-            | [] -> str
-            | h :: t when Char.IsUpper h -> String (Char.ToLower h :: t |> List.toArray)
-            | _ -> str
+        if isNull str then null else 
```

Comment:
What's the value of all of this `null` preservation that we're doing? If we want strings and we get null we could treat it as an empty string, if we get `null` when we wanted an array then `[||]`, etc.